### ISSUE TITLE
Leading incorrectly defined

### ISSUE THREAD

**Issue by andrewragy** [typst/typst#1221]
In typography, leading is the distance between 2 baselines of text as shown here:
![Leading1](https://github.com/typst/typst/assets/39835161/6fe35b57-c8ba-4440-898b-abc46e8bf89c)

However, in typst, leading is the space between the bottom of a line and the top of the next line; as shown in this picture which was generated using this code, with zero leading: 

`#set text(size: 20pt)
#set par(leading: 0pt)
#lorem(35)`

![Screenshot 2023-05-16 185956](https://github.com/typst/typst/assets/39835161/5a81a079-156a-4ba1-bd7b-c6fc783e0581)

**Comment by andrewragy** [comment]
Ok, thanks for the clarification.

**Comment by hurzl** [comment]
If leading means inserted space (lead) between lines, why do lines overlap when I set leading to zero?

**Comment by Enivex** [comment]
> If leading means inserted space (lead) between lines, why do lines overlap when I set leading to zero?

Because the bounding box of text isn't exact, depending on what `top-edge` and `bottom-edge` are set to

**Comment by hurzl** [comment]
leading is by default defined to 0.65 em (according to the docs) and I get zero "baselineskip" when I set leading to -0.65 em. Why has this to do with top-edge and bottom-edge?

**Comment by Enivex** [comment]
> leading is by default defined to 0.65 em (according to the docs) and I get zero "baselineskip" when I set leading to -0.65 em. Why has this to do with top-edge and bottom-edge?

Because leading is by definition the definition between the bottom edge of one line and the top edge of the next.

**Comment by hurzl** [comment]
Ok, but then if you want to follow the "pre-digital meaning" of leading, you should define top-edge to "ascender" and bottom-edge to "descender".
If I want to define an exact baselineskip as it is now, I have to set them both to zero and set the leading to the baselineskip I want.

**Comment by woerndl** [comment]
TL;DR: Typst currently produces a line height of 1.2 (baseline to baseline) for the 0.5em "leading" setting. If that is "incorrect" may be up for debate, but at the very least it is weird and not intuitive by modern standards. So unless Typst wants to dethrone Word for its funny line height handling, this should get fixed.

---

The term "leading" actually refers to the lead (as in lead, the metal) that typesetters used to space out lines. Following this you could argue it is the distance from the descender (e.g., the lowest point of ")", "y" or "g" etc.) to the ascender of the next line (often "I", "f", or ")"). At least this is the definition that Sheldon Cooper would give in "Fun with Flags".

It used to look like this:
![leading](https://github.com/user-attachments/assets/dff86dee-a5cb-476a-b631-8e83052c05e6)
Image Source: http://georgi3porgi3.blogspot.com/2017/07/typography-tangles-leading-kerning-what.html

That being said… I have probably used around 20-30 typesetting and layout tools in the last +10 years, and not a single one uses leading this way. Most sensible tools use a dimensionless factor, so 1.5 = 1.5em = 150% (baseline to baseline, in CSS dimensionless and percentage are not _exactly_ the same, but the [same for the same font size](https://developer.mozilla.org/en-US/docs/Web/CSS/line-height#basic_example) – different story). Microsoft Word [does something odd](https://superuser.com/questions/1784015/ms-word-how-to-get-the-height-occupied-by-a-line-of-text-in-a-specific-font-and) unless you select the "line spacing exactly" option, but then again, I would NOT (!) consider Word a "sane" tool.

The [typst documentation](https://typst.app/docs/reference/model/par/#parameters-leading) currently states that leading is "The spacing between lines," which is a bit vague… if we're talking about baselines (which is now the common understanding outside Sheldon Coopers "Fun with Flags"), then the documentation is indeed incorrect.

BUT… on top of the pedantry:

```typ
#set text(font: "Arial", 10pt)
#set par(leading: 0.5em)

This results\
in a line height of 1.2 instead of 1.5 (baseline to baseline, as in HTML, Illustrator and Figma)
```

If there is a good reason for this, I'm eager to learn it. But what currently happens is that 40% (?) of the leading value gets added between the bottom and top of the next line. So the term "incorrectly" might be subjective, but I'd say it is at least very, very odd.

There is probably a reason this gets mentioned over #159 over #4224 and over #5225 (my own, indirectly related, issue) and [over again](https://forum.typst.app/t/whats-the-equivalent-of-ms-words-1-5-line-spacing/1057).

**Comment by laurmaedje** [comment]
@woerndl See https://github.com/typst/typst/issues/4224#issuecomment-2423254670.

**Comment by woerndl** [comment]
@laurmaedje Thanks, Laurenz! By the way, congratulations on 0.12. Fantastic release—keep up the great work! 🎉
